### DISCUSSION TITLE
Jquery init not working anymore after Livewire page updated (search, filter with checkbox)

### DISCUSSION THREAD

**Discussion by giantads** [livewire/livewire#7525] [1. Help]
Hi, I have a livewire 3 application and everything works fine, until I check my first checkbox filter or hit the search button. After that I can't initialize any jquery script or plugins again. Only if i hit refresh. I hope somebody could help me. I already tried all the stackoverflow topic suggestions, I use $this->dispatch('contentChanged');, Now I need a script what loads jqurey code or plugin if anything is changed in the page. Please help me! :)

I tried also the following, with zero success
public function updateProducts()
    {
        $this->render();
        $this->dispatch('contentChanged');
    }
    
    @push('scripts')
    <script>
        window.addEventListener('contentChanged', event => {
            //Jquery code here
        });
    </script>
@endpush

**Comment by joshhanley** [comment]
@giantads personally I wouldn't be using jquery with Livewire due to these issues, instead I would be using Alpine.

But if you had to, you will probably need to listen for one of Livewire's request lifecycle javascript hooks to tell jquery to refresh once Livewire has finished. Have a look here for details https://livewire.laravel.com/docs/javascript#request-hooks

Hope this helps!

**Comment by giantads** [comment]
Thank you @joshhanley, this two lines below, have done the magic :)

document.addEventListener('livewire:init', () => {
    Livewire.hook('element.init', () => {

**Reply by joshhanley** [reply]
@giantads no worries! 🙂

**Reply by sujitProshore** [reply]
Hi @giantads 
I am facing exactly same issue with livewire 3,4 but i am using plain javascript. Can you share code snippet on where you used the event trigger in component php class and javascript snippet on how you used `document.addEventListener('livewire:init', () => {` and `Livewire.hook('element.init', () => {`. It would be really helpful as this is my frist project in livewire and struggling with this issue from 4/5 days.

**Reply by giantads** [reply]
Hey @sujitProshore

For me the script below has done the trick to init and rerun my jquery code every time that something has changed on the page.

```
<div> <!-- livewire open tag -->

<!-- YOUR CONTENT -->


<script src="https://code.jquery.com/jquery-3.7.1.min.js"   integrity="sha256-/JqT3SQfawRcv/BIHPThkBvs0OEvtFFmqPF/lYI/Cxo="   crossorigin="anonymous"></script>

<script>
    document.addEventListener('livewire:init', () => {
        Livewire.hook('element.init', () => {
            $('*[data-background-color]').each(function() {
                $(this).css({
                    'background-color': $(this).data('background-color')
                });
            });
        })
    })
</script>

</div> <!-- livewire close tag -->
```

By the way if you started to learn Livewire I suggest you @PovilasKorop  Livewire kit, you can find good, real life examples in it. That pack helped me a lot to understand livewire and its usage in different situations.

**Reply by sujitProshore** [reply]
Hi @giantads 
Thanks for the reply. Do we need to dispatch any event from our class component? I tried your method but it only works on first page load, when i do checkbox filter 'element.init' event fires only on second click(first click doesn't fire event, but second uncheck/click fires event) but still it doesn't work. Am i missing something here? Here is my code snippet:

```
<div>
  <!-- Some html contents>
  
</div>
@push('scripts')
    <script src="{{ asset('noUiSlider/dist/nouislider.js') }}"></script>
    <script src="{{ asset('/js/glider.min.js') }}"></script>
    <script>
         document.addEventListener('livewire:init', () => {
                    console.log('Hi');
                    Livewire.hook('element.init', () => {
                        console.log('Here');
                        document.querySelectorAll('.slider').forEach(function(sliderContainer) {
                            new Glider(sliderContainer, {
                                slidesToShow: 1,
                                loop: true,
                                slidesToScroll: 1,
                                draggable: true,
                                rewind: true,
                                duration: 0.5,
                                dots: sliderContainer.parentNode.querySelector('.pagination'),
                                arrows: {
                                    prev: sliderContainer.parentNode.querySelector('.slider-prev'),
                                    next: sliderContainer.parentNode.querySelector('.slider-next')
                                }
                            });
                        });
                    });
                });
    </script>
@endpush

```

**Reply by sujitProshore** [reply]
To those who are looking to use JS script in livewire pages, Don't use it. Can't figure out for my life when to re initialize our JS scripts after all DOM has been finished loading in the page, tried every livewire event and hooks available without success.

However, Got my carousel working with https://tw-elements.com/docs/standard/components/carousel/ with it's package import in my app.js.
